src/main/java/com/google/sps/utilities/GetStudentsService.java

```diff
@@ -0,0 +1,75 @@
+// Copyright 2019 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.google.sps.utilities;
+
+import com.google.sps.data.SampleData;
+import com.google.sps.data.Tutor;
+import com.google.sps.data.Student;
+import com.google.sps.data.TimeRange;
+import com.google.sps.data.TutorSession;
+import com.google.appengine.api.datastore.DatastoreService;
+import com.google.appengine.api.datastore.DatastoreServiceFactory;
+import com.google.appengine.api.datastore.Key;
+import com.google.appengine.api.datastore.KeyFactory;
+import com.google.appengine.api.datastore.Entity;
+import com.google.appengine.api.datastore.EntityNotFoundException;
+import com.google.appengine.api.datastore.PreparedQuery;
+import com.google.appengine.api.datastore.Query;
+import com.google.appengine.api.datastore.Query.FilterOperator;
+import com.google.appengine.api.datastore.Query.FilterPredicate;
+import com.google.appengine.api.datastore.Query.Filter;
+import java.lang.String;
+import java.util.Collection;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Calendar;
+import com.google.gson.Gson;
+
+/** Accesses datastore to get students that have had a tutoring session with the given tutor. */ 
+public final class GetStudentsService {
+
+    /**
+    * Gets all students that the given userId was or will be a tutor for.
+    * @return ArrayList<Student>, empty list if no students
+    */
+    public ArrayList<Student> getStudentsForTutor(String tutorId) {
+        DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+        Query studentQuery = new Query("Student");
```

Comment:
Don't know if this is necessarily easier but I'm thinking about if we have a lot of student users, this might get really slow because we would have to look through all of the students to find matching tutor ids. We could try to query TutorSessions by the tutorId and get the corresponding studentId for each session to get all the students.

Reply:
Yes, my first instinct was to look at the TutorSessions only and get the corresponding studentIds. The issue I found then was that  I'd have to look at all students to see which students have ids that match the ids  in the list of studentIds and then get their information such as (name, email,..). So this turned out to be even slower because I would not only look at all students but I'd also look at all TutorSessions.